docs/advanced_reference/developer_tooling/benchmarking.md

```diff
@@ -1,23 +1,34 @@
-# Benchmarks
+The benchmarking tool allows measuring the performance of an agent service. The tool is accessed through the class
 
-The benchmarking tools allow measuring the performance of an agent service realised on the {{open_autonomy}} framework.
+```python
+packages.valory.skills.abstract_round_abci.models.BenchmarkTool
+```
 
-## How to use benchmarking tools
+The benchmarking information is saved upon calling the method `BenchmarkTool.save()`. This function call is typically executed within the `async_act()` method of a behaviour. Therefore, the sequence of events that should occur so that the benchmarking information is saved correctly is as follows:
 
-1. Setup agent runtime environment in dev mode using `autonomy deploy build KEYS --dev`.
+1. A behaviour executes the `BenchmarkTool.save()` method.
+2. The agent service reaches the `reset_and_pause` round, and the Tendermint data is dumped.
+3. Stop the service execution after, at least, steps 1. and 2. have occurred in that sequence.
 
-2. Run agents for `1` period and wait until the round where the skill is configured to call the `BenchmarkTool.save` method. (You can also run agents for `N` periods if you want more data).
+The benchmarking data will be stored in the folder `<service_folder>/abci_build/persistent_data/benchmarks`.
 
-3. Point the aggregation script to the directory containing the benchmark data. This will generate a `benchmarks.html` file containing benchmark stats in your current directory.
+## Use the benchmarking tool
 
-## Example usage
+1. **Run the service.** Build and run the agent service. You can build and run the service either in [normal mode](../../guides/deploy_service.md#local-deployment) or in [dev mode](./dev_mode.md#build-and-run-an-agent-service-in-dev-mode).
 
-Run the `oracle/price_estimation` service as explained [here](../../demos/price_oracle_intro.md).
+2. **Wait for service execution.** Wait until the service has completed, at least, one period before cancelling the execution. That is, wait until the `reset_and_pause` round has occurred at least once. This is required because the Tendermint server will only dump Tendermint data when it reaches that state. Once you have a data dump, you can stop the local execution by pressing `Ctrl-C`.
```

Comment:
yes this is better, should come earlier (or alternatively add something like: (see below)). However, although knowing that the Tendermint data is dumped is during this moment is useful for developer, I don't see how it related to the `BenchmarkTool`.

Reply:
Fixed.